### ISSUE TITLE
In goal terminology, be consistent in usage of "ID" instead of "index" 

### ISSUE THREAD

**Issue by algoanne** [algorand/go-algorand#3671]
**Context**
I create an app using `goal app create` and it tells me "Created app with app index 2". A minute later I go to call the app using goal and the command has `--app-id`. This may seem minor, but this inconsistent terminology can cause confusion and should be cleaned up. 

**What are you looking for?**
find all occurrences of referring to app "index" in goal outputs, and update them to "ID" instead.

**Comment by barnjamin** [comment]
Same sort of issues with all our docs/api models

**Comment by FrankSzendzielarz** [comment]
Agreed!!!

**Comment by tzaffi** [comment]
@algoanne 
Before I proceed with investigating/fixing this, I want to make sure that we're all on board with changing `goal` outputs for existing commands and breaking any depending usage patterns such as:

```sh
goal app create ... | grep index | tail -n 1 | cut -d "," -f1 | awk '{print $2}'
```

**Comment by algoanne** [comment]
@tzaffi you're right. Worth discussing how we approach these changes before we do this.

**Comment by algoanne** [comment]
this is one of the few goal commands people might actually be using the output of programmatically, to get the created app id. 

One thought: could have an option to just output the created app id instead of the text (e.g. succint), if someone is using the output programmatically instead of reading it as a human. 

Conclusion of our internal discussion is that we recognize this might break a few people's code, but it's worth doing for everyone using algorand. We are probably also breaking our own e2e tests with this, so they will need to be updated as part of this work. 

This work should also include id vs. index for assets.

**Comment by winder** [comment]
@algoanne one idea [that has been proposed before](https://github.com/algorand/go-algorand/issues/4749) is to add ` --output-type=JSON` to some/all goal commands as a way to get machine readable output in place of the human readable output.
